kod/object/active/holder/nomoveon/battler/player/user.kod

```diff
@@ -4533,6 +4557,125 @@ messages:
+         {
+            plClassOrderPreferences = $;
+            Send(self,@MsgSendUser,#message_rsc=inventory_sort_cleared);
+            return;
+         }
+
+         if StringEqual(String,"inventory sort report")
+         {
+            Send(self,@MsgSendUser,#message_rsc=inventory_sort_report_header);
+
+            lSortInfo = Send(SYS,@GetSortInfo);
+
+            for i in plClassOrderPreferences
+            {
+               bFound = FALSE;
+               for n in lSortInfo
+               {
+                  if Nth(n,2) = i
+                  {
+                     Send(self,@MsgSendUser,#message_rsc=inventory_sort_report,
+                                      #parm1=Nth(n,1));
+                     bFound = TRUE;
+                     break;
+                  }
+               }
+               
+               if bFound
+               {
+                  continue;
+               }
+               
+               oTemplateItem = Send(SYS,@FindTemplateItemByClass,
+                                        #cClass=i);
+               
+               if oTemplateItem <> $
+               {
+                  Send(self,@MsgSendUser,#message_rsc=inventory_sort_report,
+                                      #parm1=Send(oTemplateItem,@GetName));
+               }
+            }
+            
+            return;
+         }
+
+         if StringContain(String,"inventory sort add ")
+         {
+            StringSubstitute(String,"inventory sort add ","");
+            
+            lSortInfo = Send(SYS,@GetSortInfo);
+            
+            for i in lSortInfo
+            {
+               if StringEqual(String,First(i))
+               {
+                  if plClassOrderPreferences <> $
+                     AND FindListElem(plClassOrderPreferences,Nth(i,2)) <> 0
+                  {
+                     Send(self,@MsgSendUser,
+                               #message_rsc=inventory_already_in_list);
+                     return;
+                  }
+               
+                  if plClassOrderPreferences = $
+                  {
+                     plClassOrderPreferences =
+                          Cons(Nth(i,2),plClassOrderPreferences);
+                  }
+                  else
+                  {
+                     plClassOrderPreferences =
+                          ConsEnd(Nth(i,2),plClassOrderPreferences);
+                  }
+                          
+                  Send(self,@MsgSendUser,#message_rsc=inventory_class_added,
+                                         #parm1=Nth(i,1));
+                  return;
+               }
+            }
+            
+            oTemplateItem = Send(SYS,@FindTemplateItemByName,#string=String);
+            
+            if oTemplateItem <> $
+            {
+               if plClassOrderPreferences = $
+               {
+                  plClassOrderPreferences =
+                       Cons(Nth(i,2),plClassOrderPreferences);
```

Comment:
plClassOrderPreferences = Cons(Nth(i,2),plClassOrderPreferences);

should be

plClassOrderPreferences = Cons(GetClass(oTemplateItem),plClassOrderPreferences);